torch/onnx/symbolic_opset13.py

```diff
@@ -437,26 +437,46 @@ def reduce(g, *args, **kwargs):
         @symbolic_helper.parse_args("v", "none")
         @_beartype.beartype
         def reduce_nodim(g, self, dtype):
+            dtype_onnx = None
             if dtype.node().kind() == "onnx::Constant":
                 dtype = symbolic_helper._get_const(dtype, "i", "dtype")
+                dtype_onnx = _type_utils.JitScalarType(dtype).onnx_type()
                 self = g.op(
-                    "Cast", self, to_i=_type_utils.JitScalarType(dtype).onnx_type()
+                    "Cast", self, to_i=dtype_onnx
                 )
             elif dtype.node().kind() != "prim::Constant":
                 return symbolic_helper._unimplemented(name, "dtype", dtype)
-            return symbolic(g, self)
+            result = symbolic(g, self)
+            if dtype_onnx is not None:
+                result_dtype_scalar = result.type().scalarType()
```

Comment:
This can cause segfault/assert crashes when model is `torch.jit.script`.
To prevent this, use exclusively `JitScalarType`'s *public* APIs to extract type information from `torch._C.Value` nodes in a safe way.

Reply:
thank you I will make that change.